synapse/rest/client/v2_alpha/account.py

```diff
@@ -158,7 +159,64 @@ def __init__(self, hs):
             hs (synapse.server.HomeServer): server
         """
         super(PasswordResetSubmitTokenServlet, self).__init__()
-        self.hs = hs
+        self.config = hs.config
+        if self.config.threepid_behaviour_email == ThreepidBehaviour.LOCAL:
+            (self.confirmation_email_template,) = load_jinja2_templates(
+                self.config.email_template_dir,
+                [self.config.email_password_reset_template_confirmation_html],
+            )
+
+    async def on_GET(self, request, medium):
+        # We currently only handle threepid token submissions for email
+        if medium != "email":
+            raise SynapseError(
+                400, "This medium is currently not supported for password resets"
+            )
+        if self.config.threepid_behaviour_email == ThreepidBehaviour.OFF:
+            if self.config.local_threepid_handling_disabled_due_to_email_config:
+                logger.warning(
+                    "Password reset emails have been disabled due to lack of an email config"
+                )
+            raise SynapseError(
+                400, "Email-based password resets are disabled on this server"
+            )
+
+        sid = parse_string(request, "sid", required=True)
+        token = parse_string(request, "token", required=True)
+        client_secret = parse_string(request, "client_secret", required=True)
```

Comment:
the call to `assert_valid_client_secret(client_secret)` should probably happen here rather than later?

Reply:
Very good point - that would be annoying...